exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -56,6 +58,12 @@
   private final boolean readNumbersAsDouble;
 
   /**
+   * Collection for tracking nullable fields during reading
+   * and storing them for creating default typed vectors
+   */
+  private final Set<String> nullableFields  = Sets.newLinkedHashSet();
```

Comment:
I don't see the reason you need it to be a LinkedHashSet. And if there is no particular reason for it, making it LinkedHashSet but not HashSet adds extra cost.